### PR TITLE
MES-3832 WRTC Vehicle Checks validation

### PR DESCRIPTION
**Pull request by mgoldson**
## Description
Displays a validation error if all of the vehicle checks questions have not been answered.
https://jira.dvsacloud.uk/browse/MES-3845

## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [x] PR link added to JIRA ticket
- [x] One review from each scrum team
- [] Squashed commit contains the JIRA ticket number

## Screenshots (optional)
Simulator Screen Shot - iPad Pro (10 5-inch) - 2019-11-25 at 12 58 40

### CODE REVIEW

File: src/pages/waiting-room-to-car/cat-be/components/vehicle-checks/__tests__/vehicle-checks.spec.ts [hunk 1]
```diff
@@ -144,6 +139,69 @@ describe('VehicleChecksCatBEComponent', () => {
...
+      const formBuilder: FormBuilder = new FormBuilder();
+      component.formGroup = formBuilder.group({
+        vehicleChecksSelectQuestions: null,
+      });
+      spyOn(component, 'everyQuestionHasOutcome').and.returnValue(false);
+      spyOn(component, 'invalidVehicleChecks');
+      component.validateVehicleChecks();
+      expect(component.invalidVehicleChecks).toHaveBeenCalled();
+    });
+  });
```
Comment (rossfauldsbjss) [inline src/pages/waiting-room-to-car/cat-be/components/vehicle-checks/__tests__/vehicle-checks.spec.ts:160]: a bit pedantic, but feel like incompleteVehicleChecks is a much better name for this.

File: src/pages/waiting-room-to-car/cat-be/components/vehicle-checks/vehicle-checks.ts [hunk 2]
```diff
@@ -53,4 +54,27 @@ export class VehicleChecksCatBEComponent {
...
+    return { vehicleChecks: false };
+  }
+
+  validateVehicleChecks() {
+    if (!this.formControl) {
+      this.formControl = new FormControl({
+        value: 'Select questions',
+        disabled: false,
+      },
+      [this.everyQuestionHasOutcome() ? null : this.invalidVehicleChecks]);
```
Comment (rossfauldsbjss) [inline src/pages/waiting-room-to-car/cat-be/components/vehicle-checks/vehicle-checks.ts:68]: Think should create a custom validator  with the check and return and add that to the form control.

Comment (rossfauldsbjss) [reply src/pages/waiting-room-to-car/cat-be/components/vehicle-checks/vehicle-checks.ts:68]: look at pass-certificate-number.ts (validatePassCertificate method) as an example custom validator.

**Review by rossfauldsbjss — APPROVED** [review]
Couple of minor things... but looks good.

File: src/pages/waiting-room-to-car/cat-be/components/vehicle-checks/__tests__/vehicle-checks.spec.ts (hunk 1 above)
Comment (mgoldson) [reply src/pages/waiting-room-to-car/cat-be/components/vehicle-checks/__tests__/vehicle-checks.spec.ts:160]: Yep, I agree.

File: src/pages/waiting-room-to-car/cat-be/components/vehicle-checks/vehicle-checks.ts (hunk 2 above)
Comment (mgoldson) [reply src/pages/waiting-room-to-car/cat-be/components/vehicle-checks/vehicle-checks.ts:68]: Thanks, I will take a look.

**Comment by mgoldson** [comment]
Incorrect JIRA ID used in branch name. New PR created with correct JIRA ID here: https://github.com/dvsa/mes-mobile-app/pull/958

**Pull request closed without merging**